### PR TITLE
[Advanced Visibility with SQL] Adding PostgreSQL 12 schema and interface

### PR DESCRIPTION
**Pull request by rodrigozhou**
**What changed?**
Added PostgreSQL 12 schemas:
- For main Temporal storage, it's just a copy from PostgreSQL v96 schema
- For visibility:
  - v1.0, v1.1: It's a copy from v96 (to support migration).
  - v1.2: Changes to support advanced visibility:
    - Added search attributes JSON map.
    - Added generated columns to extract the search attributes fields from JSON.
    - Deleted old indices and created new ones.

**Why?**
Advanced visibility with SQL databases.

**How did you test it?**
Started temporal using PostgreSQL 12, and copied existing unit tests for PostgreSQL to use the new interface.

**Potential risks**
No.

**Is hotfix candidate?**
No.

### CODE REVIEW

**Comment by MichaelSnowden** [comment]
Could you paste a diff of the SQL files? That would make it easier to review

**Comment by rodrigozhou** [comment]
@MichaelSnowden You can check the difference in the commits. The first commit is just a copy of the v96 schema plus formatting (no actual change). The second commit has the changes in the schema, and the third commit adds the interface and configs.

File: common/persistence/sql/sqlplugin/postgresql/db.go [hunk 1]
```diff
@@ -119,9 +119,9 @@ func (pdb *db) DbName() string {
 func (pdb *db) ExpectedVersion() string {
 	switch pdb.dbKind {
 	case sqlplugin.DbKindMain:
-		return postgresqlschema.Version
+		return postgresqlschemaV96.Version
```
Comment (MichaelSnowden) [inline common/persistence/sql/sqlplugin/postgresql/db.go:122]: Will this cause issues for users who upgrade?

Comment (rodrigozhou) [reply common/persistence/sql/sqlplugin/postgresql/db.go:122]: I don't think so. When they upgrade, they'll need to update the plugin name to "postgres12", which will make use the `db_v12.go`.

**Review by alexshtin — APPROVED** [review]
We should remove 9.6 support long time ago :-(

**Pull request merged**